### PR TITLE
Rename Foreign Key enum values in VSchema  and drop `FK_` prefix

### PR DESCRIPTION
**Pull request by GuptaManan100**
## Description

This PR drops the `FK_` prefix from all the enum values of `foreignKeyMode` in the VSchema. This change would have otherwise backward incompatible, but it is okay since this field has been introduced in v18 only and that hasn't had a GA release yet. We must ensure to back port this PR before then.

## Related Issue(s)

- #12967 

## Checklist

-   [x] "Backport to:" labels have been added if this change should be back-ported
-   [x] Tests were added or are not required
-   [x] Did the new or modified tests pass consistently locally and on the CI
-   [x] Documentation was added or is not required - https://github.com/vitessio/website/pull/1621

### CODE REVIEW

File: go/test/endtoend/vtgate/foreignkey/sharded_vschema.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "sharded": true,
-  "foreignKeyMode": "FK_MANAGED",
+  "foreignKeyMode": "MANAGED",
```
Comment (harshit-gangal) [inline go/test/endtoend/vtgate/foreignkey/sharded_vschema.json:3]: will this also work in lowercase or is this case sensitive?

**Review by harshit-gangal — APPROVED** [review]

**Review by deepthi — COMMENTED** [review]
Manan, can you change all the enum values to be lowercase as well?
I'm sorry, I know we discussed this in the original PR, but it became obvious when I looked at the vschema examples for the stress tests. All other vschema properties are lower-cased, and this one sticks out.
One thing harshit mentioned yesterday was testing to see whether the values specified in the vschema json are case-sensitive. I assume the answer is yes.

**Comment by GuptaManan100** [comment]
Just verified, the properties aren't case insensitive

**Comment by GuptaManan100** [comment]
I have made the required changes and the docs changes are available at https://github.com/vitessio/website/pull/1621. Once this PR is merged, I'll update the release notes too.

**Comment by GuptaManan100** [comment]
I'm not sure why this failure - 
```
[ERROR] /vt/src/vitess.io/vitess/java/client/target/generated-sources/protobuf/java/vschema/Vschema.java:[1918,61] <identifier> expected
[ERROR] /vt/src/vitess.io/vitess/java/client/target/generated-sources/protobuf/java/vschema/Vschema.java:[1922,7] ',', '}', or ';' expected
[ERROR] /vt/src/vitess.io/vitess/java/client/target/generated-sources/protobuf/java/vschema/Vschema.java:[1922,14] '}' expected
[ERROR] /vt/src/vitess.io/vitess/java/client/target/generated-sources/protobuf/java/vschema/Vschema.java:[1926,7] invalid method declaration; return type required
```

**Comment by GuptaManan100** [comment]
I made foreignKeyMode a string in the VSchema properties to fix the Java issue 👆 .
On further research I found the issue was that `default` is a keyword in Java, so the generated code 
```java
public enum ForeignKeyMode
        implements com.google.protobuf.ProtocolMessageEnum {
      /**
       * <code>default = 0;</code>
       */
      default(0),
      /**
       * <code>disallow = 1;</code>
       */
      disallow(1),
      /**
       * <code>unmanaged = 2;</code>
       */
      unmanaged(2),
      /**
       * <code>managed = 3;</code>
       */
      managed(3),
      UNRECOGNIZED(-1),
      ;
```
doesn't work. However, if I rename the default to a different value that isn't reserved, it works fine... I'll revert the string change and change `default` instead.

**Comment by GuptaManan100** [comment]
@deepthi I have fixed it now. Can be reviewed now.

**Review by deepthi — APPROVED** [review]

**Pull request merged**